### PR TITLE
曖昧検索とソートを非同期で実施されるように修正しました

### PR DESCRIPTION
**Pull request by SekiMoemi**
ご確認お願いします。

### CODE REVIEW

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java [hunk 1]
```diff
@@ -34,17 +34,11 @@ public class ShowItemListController {
 	 */
 	@GetMapping("")
 	public String showItemList(SearchItemForm form, Model model) {
-		List<Item> itemList = showItemListService.searchByName(form.getName(), form.getSortNum());
-		// 見つけない場合、全件検索する
-		if (itemList.size() == 0) {
-			itemList = showItemListService.searchByName("", form.getSortNum());
-			model.addAttribute("searchByName", "１件もありませんでした");
-		}
+		List<Item> itemList = showItemListService.searchByName("", form.getSortNum());
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:37]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:37]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 4.

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java [hunk 2]
```diff
@@ -60,4 +54,23 @@ public String narrowingDown(Integer targetAge, Integer genre, Integer priceRange
 		model.addAttribute("itemList", itemList);
 		return "item_list :: itemFragment";
 	}
+
+	/**
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:58]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java [hunk 3]
```diff
@@ -60,4 +54,23 @@ public String narrowingDown(Integer targetAge, Integer genre, Integer priceRange
...
+	 * 曖昧検索とソートされたおもちゃ一覧画面を出力します.
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:59]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java [hunk 4]
```diff
@@ -60,4 +54,23 @@ public String narrowingDown(Integer targetAge, Integer genre, Integer priceRange
...
+	 *
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:60]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java [hunk 5]
```diff
@@ -60,4 +54,23 @@ public String narrowingDown(Integer targetAge, Integer genre, Integer priceRange
 		model.addAttribute("itemList", itemList);
 		return "item_list :: itemFragment";
 	}
+
+	/**
+	 * 曖昧検索とソートされたおもちゃ一覧画面を出力します.
+	 * 
+	 * @param form  商品の検索時に利用するフォーム
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:61]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.RequireEmptyLineBeforeBlockTagGroupCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Javadoc tag '@param' should be preceded with an empty line.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:61]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java [hunk 6]
```diff
@@ -60,4 +54,23 @@ public String narrowingDown(Integer targetAge, Integer genre, Integer priceRange
...
+	 * @param model モデル
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:62]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java [hunk 7]
```diff
@@ -60,4 +54,23 @@ public String narrowingDown(Integer targetAge, Integer genre, Integer priceRange
...
+	 * @return おもちゃ一覧画面
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:63]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java [hunk 8]
```diff
@@ -60,4 +54,23 @@ public String narrowingDown(Integer targetAge, Integer genre, Integer priceRange
...
+	 */
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:64]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java [hunk 9]
```diff
@@ -60,4 +54,23 @@ public String narrowingDown(Integer targetAge, Integer genre, Integer priceRange
...
+	@GetMapping("/searchAndSort")
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:65]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:65]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def modifier' has incorrect indentation level 8, expected level should be 2.

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java [hunk 10]
```diff
@@ -60,4 +54,23 @@ public String narrowingDown(Integer targetAge, Integer genre, Integer priceRange
...
+	public String searchAndSort(SearchItemForm form, Model model) {
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:66]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java [hunk 11]
```diff
@@ -60,4 +54,23 @@ public String narrowingDown(Integer targetAge, Integer genre, Integer priceRange
...
+		List<Item> itemList = showItemListService.searchByName(form.getName(), form.getSortNum());
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:67]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 106).

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:67]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:67]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 4.

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java [hunk 12]
```diff
@@ -60,4 +54,23 @@ public String narrowingDown(Integer targetAge, Integer genre, Integer priceRange
...
+		// 見つからない場合、全件検索する
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:68]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java [hunk 13]
```diff
@@ -60,4 +54,23 @@ public String narrowingDown(Integer targetAge, Integer genre, Integer priceRange
...
+		if (itemList.size() == 0) {
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:69]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:69]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' has incorrect indentation level 16, expected level should be 4.

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java [hunk 14]
```diff
@@ -60,4 +54,23 @@ public String narrowingDown(Integer targetAge, Integer genre, Integer priceRange
...
+			itemList = showItemListService.searchByName("", form.getSortNum());
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:70]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:70]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' child has incorrect indentation level 24, expected level should be 6.

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java [hunk 15]
```diff
@@ -60,4 +54,23 @@ public String narrowingDown(Integer targetAge, Integer genre, Integer priceRange
...
+			model.addAttribute("searchByName", "１件もありませんでした");
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:71]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:71]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' child has incorrect indentation level 24, expected level should be 6.

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java [hunk 16]
```diff
@@ -60,4 +54,23 @@ public String narrowingDown(Integer targetAge, Integer genre, Integer priceRange
...
+		}
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:72]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:72]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if rcurly' has incorrect indentation level 16, expected level should be 4.

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java [hunk 17]
```diff
@@ -60,4 +54,23 @@ public String narrowingDown(Integer targetAge, Integer genre, Integer priceRange
...
+		model.addAttribute("itemList", itemList);
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:73]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:73]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 4.

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java [hunk 18]
```diff
@@ -60,4 +54,23 @@ public String narrowingDown(Integer targetAge, Integer genre, Integer priceRange
...
+		return "item_list :: itemFragment";
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:74]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:74]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 4.

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java [hunk 19]
```diff
@@ -60,4 +54,23 @@ public String narrowingDown(Integer targetAge, Integer genre, Integer priceRange
...
+	}
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java:75]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

**Review by SekiMoemi — COMMENTED** [review]
Remaining comments which cannot be posted as a review comment to avoid GitHub Rate Limit

<details>
<summary>checkstyle</summary>

[src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java|75 col 9|](http://github.com/ecsite-project/ec-202304a/blob/83f6260fd8ceb90687160a9c3a9bdd57cb50ee08/src/main/java/jp/co/example/ecommerce_a/controller/ShowItemListController.java#L75) 'method def rcurly' has incorrect indentation level 8, expected level should be 2.
[src/main/java/jp/co/example/ecommerce_a/common/SecurityConfig.java|28|](http://github.com/ecsite-project/ec-202304a/blob/83f6260fd8ceb90687160a9c3a9bdd57cb50ee08/src/main/java/jp/co/example/ecommerce_a/common/SecurityConfig.java#L28) Line is longer than 100 characters (found 154).
[src/main/java/jp/co/example/ecommerce_a/common/SecurityConfig.java|28 col 1|](http://github.com/ecsite-project/ec-202304a/blob/83f6260fd8ceb90687160a9c3a9bdd57cb50ee08/src/main/java/jp/co/example/ecommerce_a/common/SecurityConfig.java#L28) Line contains a tab character.
[src/main/java/jp/co/example/ecommerce_a/common/SecurityConfig.java|28 col 120|](http://github.com/ecsite-project/ec-202304a/blob/83f6260fd8ceb90687160a9c3a9bdd57cb50ee08/src/main/java/jp/co/example/ecommerce_a/common/SecurityConfig.java#L28) ',' is not followed by whitespace.
[src/main/java/jp/co/example/ecommerce_a/common/SecurityConfig.java|28 col 137|](http://github.com/ecsite-project/ec-202304a/blob/83f6260fd8ceb90687160a9c3a9bdd57cb50ee08/src/main/java/jp/co/example/ecommerce_a/common/SecurityConfig.java#L28) ',' is not followed by whitespace.
</details>

**Review by yamada-kensyu — APPROVED** [review]
承認します

**Pull request merged**